src/cli/nexus-prisma.ts

```diff
@@ -26,7 +25,15 @@ generatorHandler({
   },
   // async required by interface
   // eslint-disable-next-line
-  async onGenerate({ dmmf }) {
+  async onGenerate({ dmmf, otherGenerators }) {
+    const prismaClientGenerator = otherGenerators.find((g) => g.provider.value === 'prisma-client-js')
+
+    if (!Gentime.settings.data.prismaClientLocation) {
```

Comment:
So, given that the user configuration is read _after_ (a few lines down) we don't actually need to do the check `Gentime.settings.metadata.fields.prismaClientImportId.from === 'initial'` because it will _always_ be `true`.

Aside: A quirk of the current Setset API is that our internal adjustment here will affect the metadata such that this setting looks touched, but actually it probably shouldn't be considered touched, it is just a dynamic default, really.

So here's my revised suggestion:

```suggestion
    // WARNING: Make sure this logic comes before `loadUserGentimeSettings` below 
    // otherwise we will overwrite the user's choice for this setting if they have set it.
    if (prismaClientGenerator?.output?.value) {
```